src/main/java/frc/robot/subsystems/swerve/ModuleIOReal.java

```diff
@@ -216,7 +217,11 @@ public void updateInputs(final ModuleIOInputs inputs, final List<Samples> asyncO
     inputs.odometryTimestamps =
         asyncOdometrySamples.stream().mapToDouble(s -> s.timestamp()).toArray();
     inputs.odometryDrivePositionsMeters =
-        asyncOdometrySamples.stream().mapToDouble(s -> s.values().get(drivePosition)).toArray();
+        asyncOdometrySamples.stream()
+            .map(s -> s.values().get(drivePosition))
+            .map(d -> d == null ? new NullableDouble(null) : new NullableDouble(d))
```

Comment:
You're always passing d to the constructor. Might as well just `.map(NullableDouble::new)`.